lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback, args) {
```

Comment:
I tried various constructs, used the fastest, and used style as a tie-breaker.

I timed the `new Immediate(callback, args)` version setting properties on `this` with
and without a prototype, and the current static object is 50% faster on the breadth tests
in benchmarks/timers/immediate.js (10% for the slow 4+ args case)

my experimental Immediate was (a separate test predeclared all properties as null)
```
function Immediate(callback, args) {
  this._callback = callback;
  this._argv = args;
  this._onImmediate = callback;
  this.domain = process.domain;
}
```